src/frontend/src/components/SearchBar.tsx

```diff
@@ -44,13 +49,18 @@ const StyledInputBase = styled(InputBase)(({ theme }) => ({
   }
 }));
 
-export const SearchBar = ({ placeholder = 'Search...' }: { placeholder?: string }) => {
+export const SearchBar = ({ placeholder = 'Search...', searchText, setSearchText }: SearchBarProps) => {
   return (
     <Search>
       <SearchIconWrapper>
         <SearchIcon />
       </SearchIconWrapper>
-      <StyledInputBase placeholder={placeholder} inputProps={{ 'aria-label': 'search' }} />
+      <StyledInputBase
+        placeholder={placeholder}
+        inputProps={{ 'aria-label': 'search' }}
+        value={searchText}
+        onChange={(event) => setSearchText(event.target.value)}
+      />
```

Comment:
can we add an x button on the right side of the search input to clear the contents when clicked, I feel like we have an example somewhere but I can't find it